app/src/main/proto/vss.proto

```diff
@@ -114,45 +114,60 @@ message GetKeysSummaryRequest {
   //
-  // For subsequent pages, use the value that was returned as `last_evaluated_key` in the previous
-  // page's GetKeysSummaryResponse.
-  // It will be used as exclusive_start_key to perform the new query and only keys greater than
-  // the specified last_evaluated_key will be returned in the response.
-  optional string last_evaluated_key = 3;
+  // For subsequent pages, use the value that was returned as `next_page_token` in the previous
+  // page's ListKeyVersionsResponse.
+  optional string page_token = 4;
 }
 
-message GetKeysSummaryResponse {
+message ListKeyVersionsResponse {
 
-  // Fetched versions along with the corresponding keys in the request.
+  // Fetched keys and versions.
+  // Even though this API reuses KeyValue struct, the value sub-field will not be set by the server.
   repeated KeyValue key_versions = 1;
 
-  // last_evaluated_key is a pagination token, it is the key of the item where the operation stopped,
-  // inclusive of the current result set.
-  // Use this value to start a new GetKeysSummary operation, excluding this value in the new result
-  // by specifying this last_evaluated_key in the next request.
+  // next_page_token is a pagination token, used to retrieve the next page of results.
+  // Use this value to query for next_page of paginated ListKeyVersions operation, by specifying
+  // this value as the `page_token` in the next request.
   //
-  // If last_evaluated_key is empty, then the "last page" of results has been processed and there is
-  // no more data to be retrieved.
+  // If next_page_token is empty (""), then the "last page" of results has been processed and
+  // there is no more data to be retrieved.
   //
-  // If last_evaluated_key is not empty, it does not necessarily mean that there is more data in the
+  // If next_page_token is not empty, it does not necessarily mean that there is more data in the
   // result set. The only way to know when you have reached the end of the result set is when
-  // last_evaluated_key is empty.
+  // next_page_token is empty.
   //
   // Caution: Clients must not assume a specific number of key_versions to be present in a page for
   // paginated response.
-  optional string last_evaluated_key = 2;
+  optional string next_page_token = 2;
 
   // global_version is a sequence-number/version of the whole store.
   //
-  // global_version is only returned in response for the first page of GetKeysSummaryResponse.
+  // global_version is only returned in response for the first page of the ListKeyVersionsResponse
+  // and is guaranteed to be read before reading any key-versions.
+  //
+  // In case of refreshing complete key-version view on the client-side, correct usage for
+  // the returned global_version is as following:
+  //   1. Read global_version from the first page of paginated response, store it as local_variable.
+  //   2. Update all the key_versions on client-side from the paginated response.
```

Comment:
Does this also mean calling `GetObject` for each key? It isn't clear from my reading of this that it is required though it seems implied from my understanding of global versioning.

Reply:
see my other comment above about globally consistent snapshot

Reply:
Yes, 
lets say client makes a put operation and it fails due to conflictException. Now client wishes to refresh their view of storage. 
This can be done  by computing the diff in versions and downloading the updated keys in lazy or eager fashion.

Since only few of the keys need to be updated to ensure fully-updated view, only those keys will be fetched.